src/github.com/matrix-org/dendrite/roomserver/storage/storage.go

```diff
@@ -38,7 +38,7 @@ func (d *Database) SetPartitionOffset(topic string, partition int32, offset int6
 }
 
 // StoreEvent implements input.EventDatabase
-func (d *Database) StoreEvent(event gomatrixserverlib.Event) error {
+func (d *Database) StoreEvent(event gomatrixserverlib.Event, authEventNIDS []int64) error {
```

Comment:
The event type comes from gomatrixserverlib, The concept of NIDs exists only within the roomserver, so the list of NIDs doesn't really belong on the Event type.

The auth events roughly correspond to the auth_events key in the event JSON, but there exist events in the wild that are missing necessary events from the auth_events key, so we keep the concepts separate in the API so that we can fix the list for those events.